---
title: "Toxo_enrichment_tables"
author: "JesusAV"
date: "2025-07-29"
output: html_document
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T, message = F, warning = F)
```

```{r Packages, echo=F}
library(tidyverse)
library(gridExtra)
library(knitr) 
library(dplyr)
library(ggstats)
library(ggplot2)
library(ggbeeswarm)
library(corrplot)
library(colorspace)

```


```{r raw_table}
# Load information of raw Motif matches predictions
raw_matches <- read_tsv(file="/Data/ALIAS_MotifMatches_list.txt")
raw_matches %>% select(Protein_ID) %>% unique() %>% nrow() # 8,204 unique protein IDs

# Identify ToxoDB IDs
raw_matches <- raw_matches %>% mutate(ToxoDB_ID = gsub("(\\.|-).*?$","",Protein_ID)) %>% mutate(Motif_type = gsub("_.*?$","",Motif_Name))
raw_matches %>% select(ToxoDB_ID) %>% unique() %>% nrow()  # 8,140 ToxoDB proteins

raw_matches <- raw_matches %>% mutate(Protein_ID=gsub("-p1$","",Protein_ID)) # clean sequence ID
raw_matches <- raw_matches %>% mutate(Motif_eSite=Motif_sSite+nchar(Motif_Instance)) # Add ending site
raw_matches <- raw_matches %>% mutate(Motif_sSite=Motif_sSite+1) # Correst start site

```


```{r PTM_raw_table}
# Load ToxoDB phosphosite table
phosphosites_raw <- read_tsv(file="/Data/ToxoDB_29072025_Phosphosites_raw.txt")
colnames(phosphosites_raw) <- c("Gene_ID","source_id", "Modified_Residues","Total_Modified_Residues", "Total_Modifications_By_Type") # Change column names for joining tables
phosphosites_raw <- phosphosites_raw %>% select(-Total_Modifications_By_Type) # Take away redundant columns
phosphosites_raw <- phosphosites_raw %>% mutate(strain=gsub("_.*?$", "", Gene_ID)) %>% filter(strain!="TGGT1") %>% select(-strain) # Remove phosphosites from other strains

max(phosphosites_raw$Total_Modified_Residues) 
phosphosites <- phosphosites_raw %>% select(-Total_Modified_Residues) %>% separate_rows(Modified_Residues, sep = ",") #separate rows to get single columns per phosphosite
phosphosites <- phosphosites %>% separate(Modified_Residues, c("aa_res", "aa_pos"), sep=":") # separate information  of position and amino acid code
#write_tsv(phosphosites, file="/Data/ToxoDB_29072025_Phosphosites.tsv") 

```


```{r phosphosites_survey}
# Combine phosphosite table with motif matches
matches_phosphosite_survey <- raw_matches %>% filter(ToxoDB_ID %in% phosphosites$Gene_ID) %>% #focus on proteins with actual modification
  select(ToxoDB_ID, Motif_Name, Match_N, Motif_Instance, Motif_sSite) %>% # select minimal information to identify the matches
  mutate(Motif_eSite=Motif_sSite+nchar(Motif_Instance)-1) 
matches_phosphosite_protein <- matches_phosphosite_survey %>% select(ToxoDB_ID) %>% unique() # List of proteins with modification

matches_phosphosite <- matches_phosphosite_survey[0,] #empty dataframe with compatible columns
matches_phosphosite$Modification_n <- "" #Empty column for saving modification count
for(i in 1:nrow(matches_phosphosite_protein)){
  matches_phosphosite_sample <- matches_phosphosite_survey %>% filter(ToxoDB_ID==unlist(matches_phosphosite_protein[i,1])) # Get matches for the given protein
  phosphosites_sample <-  phosphosites %>% filter(Gene_ID==unlist(matches_phosphosite_protein[i,1])) # Get modification of the given protein
  matches_phosphosite_sample$Modification_n <- 0 # Set 0 as default initial value
  
  for(j in 1:nrow(matches_phosphosite_sample)){
    # count the number of modifications within the motif
    matches_phosphosite_sample[j,7] <- phosphosites_sample %>%  
      filter(Gene_ID==unlist(matches_phosphosite_sample[j,1]) & 
               aa_pos >= unlist(matches_phosphosite_sample[j,5]) &
               aa_pos <= unlist(matches_phosphosite_sample[j,6])) %>% nrow()
  }
  
  matches_phosphosite <- rbind(matches_phosphosite, matches_phosphosite_sample) # Add temporary dataset to larger one
  
}

matches_phosphosite <- matches_phosphosite %>% filter(Modification_n>0) # Take away matches without modifications

#write_tsv(matches_phosphosite, file="/Users/jesusav/Desktop/July2025_MotifMatches_phosphosites.tsv")


```


```{r UniProt_domains_raw_table}
# Load domain boundaries information from UniProt
uniprot_domains_raw <- read_tsv(file="/Data/uniprotkb_taxonomy_id_508771_2025_07_29.tsv")
colnames(uniprot_domains_raw) <- gsub(" ", "_", colnames(uniprot_domains_raw)) # Remove spaces from column names

uniprot_domains <- uniprot_domains_raw %>% rename(Domain=`Domain_[FT]`) %>% filter(!is.na(Domain)) # Rename domain column name and keep proteins with actual domains
uniprot_domains <- uniprot_domains %>% separate_rows(Domain, sep = "; DOMAIN ") %>% mutate(Domain=gsub("DOMAIN ","", Domain)) # separate protein domain per single row
uniprot_domains <- uniprot_domains %>% separate(Domain, c("range", "note", "evidence"), sep="; /") # Separate single domain information
uniprot_domains <- uniprot_domains %>% mutate(note=gsub("note=","",gsub("\"", "",note)), evidence=gsub("evidence=.*?\\|","",gsub("\"", "",evidence))) # take away extra/redundant characters
uniprot_domains <- uniprot_domains %>% mutate(range=gsub("\\.\\.","-",range), Gene_Names=gsub("^.*?(TGME49)","\\1",Gene_Names)) # Get ToxoDB IDs
uniprot_domains <- uniprot_domains %>% separate_rows(Gene_Names, sep = " ") # Separate ToxoIDs as single rows

#write_tsv(uniprot_domains %>% select(-Protein_names), file="/Data/UniProt_29072025_domains.tsv")
```


```{r domains_survey}
# Determine domain overlaps with motif matches
domains <- uniprot_domains %>% select(Gene_Names, Entry, note, range) # Get minimal information for overlap 
domains <- domains %>% separate(range, c("sDomain","eDomain"), sep="-", convert = T) # separate start and end of domain


matches_domain_survey <- raw_matches %>% filter(ToxoDB_ID %in% uniprot_domains$Gene_Names) %>%  #get matches from proteins with annotated domains
  select(ToxoDB_ID, Motif_Name, Match_N, Motif_Instance, Motif_sSite) %>% 
  mutate(Motif_eSite=Motif_sSite+nchar(Motif_Instance)-1)
matches_domain_protein <- matches_domain_survey %>% select(ToxoDB_ID) %>% unique()

matches_domains <- left_join(matches_domain_survey, domains %>% rename(ToxoDB_ID=Gene_Names)) #combine matches with domain boundaries using ToxoDB ids
matches_domains$Domain_n <- 1*with(matches_domains, ( # Make a logical vector on the overlap of the motif within the domain
                                 (Motif_sSite >= sDomain & Motif_sSite <= eDomain ) | 
                                   (Motif_eSite >= sDomain & Motif_eSite <= eDomain) ) ) 

matches_domains$Domain_n1 <- 1*with(matches_domains,   (Motif_sSite >= sDomain & Motif_sSite <= eDomain )  )

# as a match can overlap with multiple domains, group by motif match and count domain overlaps
matches_domains <- matches_domains %>% select(-Entry, -note ) %>% group_by(ToxoDB_ID, Motif_Name, Motif_Instance, Match_N, Motif_sSite, Motif_eSite) %>% summarise(Domain_overlap=sum(Domain_n)) %>% ungroup()

matches_domains <- matches_domains %>% filter(Domain_overlap>0) #Get only matches with domain overlaps

#write_tsv(matches_domains, file="/Data/ALIAS_MotifMatches_domains.tsv")


```

```{r raw_pLDDT}
# Get UniProt IDs to be mapped to ToxoDB ID
uniprot_ids <- read_tsv(file="/Data/Genes_UniProt.txt") 
colnames(uniprot_ids) <- c("ToxoDB_ID", "Protein_ID", "Description", "UPID") 
uniprot_ids <- uniprot_ids %>% separate_rows(UPID, sep=",") %>% rename(UniProt_ID=UPID)%>% filter(!is.na(UniProt_ID))

# Load all pLDDT values from Toxoplasma AF single protein models
pLDDT_values <- read_table(file="/Data/all_models_pLDDT_values.txt", col_names = c("File", "AA", "Chain", "residue_n","plDDT")) 
pLDDT_values <- pLDDT_values %>% mutate(UniProt_ID=gsub("-.*$","",gsub("AF-", "",File))) %>% select(-File, -Chain) # Get Protein IDs from model names
pLDDT_values <- left_join(pLDDT_values, uniprot_ids %>% select(-Description)) # add ToxoDB IDs
pLDDT_values <- pLDDT_values %>% mutate(Protein_ID=gsub("\\.R",".P",Protein_ID))  # edit IDs

matches_pLDDT <- raw_matches  %>% select(Protein_ID, Motif_Name,Motif_sSite, Motif_eSite) %>% mutate(avg_pLDDT=NA) # Create table for matches pLDDT info
matches_pLDDT <- matches_pLDDT %>% mutate(Protein_ID=gsub("-p1","",Protein_ID))  # edit match ToxoDB ID

proteins_ID <- pLDDT_values %>% filter(!is.na(Protein_ID)) %>% select(Protein_ID) %>% unique() %>% unlist() %>% as.vector() # Focus on protein with actual AF single protein models
matches_pLDDT_all <- matches_pLDDT[0,] # initialise matches table for adding pLDDT

for(i in 1:length(proteins_ID)){
  protein_motifs <- matches_pLDDT %>% filter(Protein_ID==proteins_ID[i]) # get the motif matches of the given protein
  protein_pLDDTs <- pLDDT_values %>% filter(Protein_ID==proteins_ID[i]) # get the pLDDT residues values of the given protein
  for(j in 1:nrow(protein_motifs)){
     protein_motifs[j,5] <- protein_pLDDTs %>% filter(residue_n>=unlist(protein_motifs[j,3]) & residue_n<= unlist(protein_motifs[j,4])) %>% select(plDDT) %>% unlist() %>% mean() # get the pLDDT values from the motif ranges and summarise them 
  }
  matches_pLDDT_all <- rbind(matches_pLDDT_all, protein_motifs) # add motif pLDDT averages to the all matches tables
}

#write_tsv(matches_pLDDT_all, file="/Data/ALIAS_MotifMatches_pLDDT.tsv")

```


```{r dssp_accessibility}
# Map of residue total accessibilities for standardisation
Accessibility_map <- c(
  A=129, R=274, N=195, D=193, C=167,
  E=223, Q=225, G=104, H=224, I=197,
  L=201, K=236, M=224, F=240, P=159,
  S=155, T=172, W=285, Y=263, V=174
)

# Load DSSP accessibility values of all AF single protein models
dssp_values <- read_table(file="/Data/all_model_dssp_values.txt", col_names = c("File", "residue_n", "Chain", "AA","total_accessibility")) 
dssp_values <- dssp_values %>% mutate(UniProt_ID=gsub("-.*$","",gsub("AF-", "",File))) %>% select(-File, -Chain) # Get Protein IDs from model names
dssp_values <- left_join(dssp_values, uniprot_ids %>% select(-Description)) # add ToxoDB IDs
dssp_values <- dssp_values %>% mutate(Protein_ID=gsub("\\.R",".P",Protein_ID)) # edit IDs

dssp_values <- dssp_values %>% mutate(norm_accessibility=(total_accessibility/Accessibility_map[AA])) # Determine Relative Surface accessibility (RSA)

matches_acc <- raw_matches  %>% select(Protein_ID, Motif_Name,Motif_sSite, Motif_eSite) %>% mutate(avg_RSA=NA)  # Create table for matches RSA info
matches_acc <- matches_acc %>% mutate(Protein_ID=gsub("-p1","",Protein_ID))  # edit match ToxoDB ID

proteins_ID <- dssp_values %>% filter(!is.na(Protein_ID)) %>% select(Protein_ID) %>% unique() %>% unlist() %>% as.vector() # Focus on protein with actual AF single protein models
matches_acc_all <- matches_acc[0,] # initialise matches table for adding RSa

for(i in 1:length(proteins_ID)){
  protein_motifs <- matches_acc %>% filter(Protein_ID==proteins_ID[i]) # get the motif matches of the given protein
  protein_accs <- dssp_values %>% filter(Protein_ID==proteins_ID[i]) # get the RSA residues values of the given protein
  for(j in 1:nrow(protein_motifs)){
     protein_motifs[j,5] <- protein_accs %>% 
       filter(residue_n>=unlist(protein_motifs[j,3]) & residue_n<= unlist(protein_motifs[j,4])) %>% 
       select(norm_accessibility) %>% unlist() %>% mean() # get the RSA values from the motif ranges and summarise them 
  }
  matches_acc_all <- rbind(matches_acc_all, protein_motifs)  # add motif RSA averages to the all matches tables
}

#write_tsv(matches_acc_all, file="/Data/ALIAS_MotifMatches_RSA.tsv")

```
